tests/framework/utils.py

```diff
@@ -49,36 +51,132 @@ class CONST:
 
 CONST = CONST()
 
+class ClusterManager(ABC):
+    EXTERNAL_CLUSTER = "EXTERNAL_CLUSTER"
 
-class KubernetesClusterManager:
+    @property
+    @abstractmethod
+    def client_dict(self):
+        pass
+
+    @abstractmethod
+    def initialize_cluster(self, kind_config=None) -> None:
+        pass
+
+    @abstractmethod
+    def cleanup(self) -> None:
+        pass
+
+    @abstractmethod
+    def upload_image():
+        pass
+
+    @abstractmethod
+    def check_cluster_exist(self) -> bool:
+        pass
+    
+    @classmethod
+    def instance(cls):
+        if cls.EXTERNAL_CLUSTER in os.environ:
+            return ExternalClusterManager()
+        else:
+            return KindClusterManager()
+
+class ExternalClusterManager(ClusterManager):
+    def __init__(self) -> None:
+        self.client_dict = {}
+
+    def client_dict(self):
+        return self.client_dict
+    
+    def cleanup(self, namespace = "default") -> None:
+        config.load_kube_config()
+        api_extensions = client.ApiextensionsV1Api()
+        apps_v1 = client.AppsV1Api()
+        custom_objects_api = client.CustomObjectsApi()
+        crds = api_extensions.list_custom_resource_definition()
+        for crd in crds.items:
```

Comment:
Custom resources should be cleaned up by the `CREvent` instead of the cluster manager.